Problems with update_repo in 1.6 when using run.sh

When I run the update_repo routine using the standard `run.sh` mode I get the following situation:

1. When updating from 1.5.X to 1.6 I first have to use `sh run.sh make update_repo`. 
2. After the initial migration is done that command does not work anymore (do you think it would be ok to still keep a makegoal that points to the new update_repo script to be a bit more backwards compatible?)
3. Then I run `sh run.sh update_repo` which mostly works, but we end up with a warning: `run.sh: line 102: syntax error near unexpected token `)'`
4. We are used to such warnings so we try to run again: `sh run.sh update_repo`. This time it is game over: `docker: Error response from daemon: invalid mode: /run/host-services/ssh-auth.sock.`.

Note: Using the odkrunner works fine:

```
odk update_repo
```

<details><summary>Log</summary>

```
(.venv) ➜  ontology git:(rename-concentration1) ✗ sh run.sh make update_repo
useradd warning: odkuser's uid 501 outside of the UID_MIN 1000 and UID_MAX 60000 range.
hp.Makefile:137: warning: overriding recipe for target 'hp.owl'
.... some stuff 
sh ../scripts/update_repo.sh
This (experimental) update script will create a new repo according to your config file. It will:
(1) overwrite your repositories Makefile, ODK sparql queries (your custom queries wont be touched) and docker wrapper (run.sh).
(2) and add missing files, if any.
INFO:root:  Copying: /tools/templates/CODE_OF_CONDUCT.md -> target/hp/CODE_OF_CONDUCT.md
.... some stuff
Repository files have been successfully copied, but no git commands have been run.
total 0
drwxr-xr-x  5 odkuser dialout 160 Mar 23 09:49 metadata
drwxr-xr-x 13 odkuser dialout 416 Mar 23 09:49 ontology
drwxr-xr-x  7 odkuser dialout 224 Mar 23 09:49 patterns
drwxr-xr-x  3 odkuser dialout  96 Mar 23 09:49 scripts
drwxr-xr-x 19 odkuser dialout 608 Mar 23 09:49 sparql
drwxr-xr-x 26 odkuser dialout 832 Mar 23 09:49 translations
total 0
drwxr-xr-x  7 odkuser dialout  224 Aug 11  2020 mappings
drwxr-xr-x  5 odkuser dialout  160 Feb 10  2021 metadata
drwxr-xr-x 52 root    root    1664 Mar 23 09:49 ontology
drwxr-xr-x  3 odkuser dialout   96 Aug 11  2020 owltools
drwxr-xr-x 12 odkuser dialout  384 Mar 23 09:34 patterns
drwxr-xr-x 18 odkuser dialout  576 Mar 23 09:34 scripts
drwxr-xr-x 73 odkuser dialout 2336 Mar 23 09:34 sparql
drwxr-xr-x 94 odkuser dialout 3008 Mar 23 08:37 translations
cp: cannot stat 'target/hp/src/scripts/update_repo.sh': No such file or directory
make: *** [Makefile:1139: update_repo] Error 1
(.venv) ➜  ontology git:(rename-concentration1) ✗ sh run.sh update_repo 
useradd warning: odkuser's uid 501 outside of the UID_MIN 1000 and UID_MAX 60000 range.
INFO:root:Applying templates
INFO:root:  Unpacking: ../../_dynamic_files
INFO:root:  Unpacking into: ../../src/sparql/hp_terms.sparql
INFO:root:  Unpacking into: ../../src/sparql/owldef-self-reference-violation.sparql
INFO:root:  Unpacking into: ../../src/sparql/taxon-range-violation.sparql
INFO:root:  Unpacking into: ../../src/sparql/iri-range-violation.sparql
INFO:root:  Unpacking into: ../../src/sparql/label-with-iri-violation.sparql
INFO:root:  Unpacking into: ../../docs/odk-workflows/index.md
INFO:root:  Unpacking into: ../../docs/odk-workflows/ContinuousIntegration.md
INFO:root:  Unpacking into: ../../docs/odk-workflows/EditorsWorkflow.md
INFO:root:  Unpacking into: ../../docs/odk-workflows/ReleaseWorkflow.md
INFO:root:  Unpacking into: ../../docs/odk-workflows/RepoManagement.md
INFO:root:  Unpacking into: ../../docs/odk-workflows/RepositoryFileStructure.md
INFO:root:  Unpacking into: ../../docs/odk-workflows/SettingUpDockerForODK.md
INFO:root:  Unpacking into: ../../docs/odk-workflows/UpdateImports.md
INFO:root:  Unpacking into: ../../docs/odk-workflows/components.md
INFO:root:  Unpacking into: ../../docs/odk-workflows/ManageDocumentation.md
INFO:root:  Unpacking into: ../../docs/odk-workflows/ManageAutomatedTest.md
INFO:root:  Unpacking into: ../../.github/workflows/docs.yml
INFO:root:Applying templates
INFO:root:Applying templates
INFO:root:  Compiling: /tools/templates/src/ontology/Makefile.jinja2 -> ../../src/ontology/Makefile
INFO:root:  Compiling: /tools/templates/src/ontology/run.sh.jinja2 -> ../../src/ontology/run.sh
INFO:root:Applying templates
INFO:root:  Copying: /tools/templates/src/sparql/class-count-by-prefix.sparql -> ../../src/sparql/class-count-by-prefix.sparql
INFO:root:  Copying: /tools/templates/src/sparql/preprocess-module.ru -> ../../src/sparql/preprocess-module.ru
INFO:root:  Copying: /tools/templates/src/sparql/basic-report.sparql -> ../../src/sparql/basic-report.sparql
INFO:root:  Copying: /tools/templates/src/sparql/synonyms.sparql -> ../../src/sparql/synonyms.sparql
INFO:root:  Copying: /tools/templates/src/sparql/simple-seed.sparql -> ../../src/sparql/simple-seed.sparql
INFO:root:  Copying: /tools/templates/src/sparql/xrefs.sparql -> ../../src/sparql/xrefs.sparql
INFO:root:  Copying: /tools/templates/src/sparql/obsoletes.sparql -> ../../src/sparql/obsoletes.sparql
INFO:root:  Copying: /tools/templates/src/sparql/subsets-labeled.sparql -> ../../src/sparql/subsets-labeled.sparql
INFO:root:  Copying: /tools/templates/src/sparql/postprocess-module.ru -> ../../src/sparql/postprocess-module.ru
INFO:root:  Copying: /tools/templates/src/sparql/terms.sparql -> ../../src/sparql/terms.sparql
INFO:root:  Copying: /tools/templates/src/sparql/edges.sparql -> ../../src/sparql/edges.sparql
INFO:root:Applying templates
INFO:root:  Compiling: /tools/templates/src/sparql/README.md.jinja2 -> ../../src/sparql/README.md
INFO:root:Applying templates
WARNING: These files should be manually migrated:
         mkdocs.yaml, .gitignore, src/ontology/catalog.xml
         (if you added a new import or component)
WARNING: Your QC workflows have not been updated automatically.
         Please update the ODK version number in .github/workflows/qc.yml
Ontology repository update successfully completed.
run.sh: line 102: syntax error near unexpected token `)'
(.venv) ➜  ontology git:(rename-concentration1) ✗ sh run.sh update_repo
docker: Error response from daemon: invalid mode: /run/host-services/ssh-auth.sock.
See 'docker run --help'.

```
</details>

> Please post your entire run.sh script… Line 102 is merely where execution stopped when the script was overwritten by the update process, it most likely has nothing to do with the subsequent error.

Script before run.sh:
https://www.dropbox.com/scl/fi/87ysstwa0uuc8q0oe8ead/run.sh.pre?rlkey=ecov8ylsfs6whar7s7nzbp48z&dl=0

Script after run.sh:
https://www.dropbox.com/scl/fi/wqlmrf58iydpmguxck4r3/run.sh.post1?rlkey=4qwq4p8dtgxqvri68xvda54cr&dl=0

> But does the updated run.sh script works fine in that configuration? The fact that you didn’t a run.sh error here is simply because the script is not overwritten while it is being executed (that’s the whole point of the ODKRunner), but that does not mean the newly generated script would not have the same problem as what you get when you run the update process without the ODKRunner.

Indeed it does not, sorry for not checking. Even if using odk runner it does not work.

> What is this “babelon2” environment?

babelon2 and venv are just python environments that happened to be active, I switched them off and tried again to no avail.

> What is the value of your SSH_AUTH_SOCK environment variable (both in your normal environment and in this “babelon2” environment)?

empty everywhere.

> I still cannot reproduce your problem here.


So I isolated the problem a little more. It seems like nothing I said before matters:

No matter how I run update_repo the first time, every command _apart_ from `sh run.sh update_repo` works fine:

```
 sh run.sh update_repo   
docker: Error response from daemon: invalid mode: /run/host-services/ssh-auth.sock.
See 'docker run --help'.
```

`make` works:

```
➜  ontology git:(master) ✗ sh run.sh make odkversion
useradd warning: odkuser's uid 501 outside of the UID_MIN 1000 and UID_MAX 60000 range.
hp.Makefile:137: warning: overriding recipe for target 'hp.owl'
Makefile:1108: warning: ignoring old recipe for target 'hp.owl'
hp.Makefile:142: warning: overriding recipe for target 'hp.obo'
Makefile:1112: warning: ignoring old recipe for target 'hp.obo'
hp.Makefile:146: warning: overriding recipe for target 'hp.json'
Makefile:1114: warning: ignoring old recipe for target 'hp.json'
hp.Makefile:160: warning: overriding recipe for target 'mirror-nbo'
Makefile:489: warning: ignoring old recipe for target 'mirror-nbo'
hp.Makefile:528: warning: overriding recipe for target 'public_release'
Makefile:1178: warning: ignoring old recipe for target 'public_release'
hp.Makefile:572: warning: overriding recipe for target 'help'
Makefile:1235: warning: ignoring old recipe for target 'help'
hp.Makefile:582: warning: overriding recipe for target 'hp-fr.owl'
Makefile:1163: warning: ignoring old recipe for target 'hp-fr.owl'
hp.Makefile:590: warning: overriding recipe for target '../translations/hp-cs-preprocessed.babelon.tsv'
Makefile:861: warning: ignoring old recipe for target '../translations/hp-cs-preprocessed.babelon.tsv'
ODK Makefile v1.6
ODK Image v1.6
ROBOT version 1.9.7
DOSDP-Tools v0.19.3
Konclude v0.7.0-1138
Jena v4.9.0
Relation-Graph v2.3.2
Ontology Access Kit v0.6.21
Ammonite REPL & Script-Runner, 2.5.9
Scala CLI version 1.5.4
SSSOM-CLI version 1.1.1
```

Also robot:

```
sh run.sh robot --version
useradd warning: odkuser's uid 501 outside of the UID_MIN 1000 and UID_MAX 60000 range.
ROBOT version 1.9.7
```

Here is the pinnacle of weird:

```
➜  ontology git:(master) ✗ sh run.sh cat /tools/robot | head -4
useradd warning: odkuser's uid 501 outside of the UID_MIN 1000 and UID_MAX 60000 range.
#!/bin/sh

## Check for Cygwin, use grep for a case-insensitive search
➜  ontology git:(master) ✗ sh run.sh cat /tools/update_repo | head -4
docker: Error response from daemon: invalid mode: /run/host-services/ssh-auth.sock.
See 'docker run --help'.
➜  ontology git:(master) ✗ odk cat /tools/update_repo | head -4
#!/bin/sh

exec /tools/odk.py update
```

It seems that there is some interaction between my computer, `run.sh` and `/tools/update_repo` which causes this.

--- a/template/src/ontology/run.sh.jinja2
+++ b/template/src/ontology/run.sh.jinja2
@@ -20,7 +20,7 @@ case "$@" in
     # during the update process
     mkdir -p tmp
     sed -n '/^set -e/,$p' $0 > tmp/$0
-    exec $SHELL tmp/$0 "$@"
+    exec /bin/sh tmp/$0 "$@"
     ;;
 esac
 

